src/pcapi/scripts/payment/user_recredit.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+import logging
+
+from dateutil.relativedelta import relativedelta
+from sqlalchemy.orm import joinedload
+
+from pcapi.core.payments import models as payments_models
+import pcapi.core.payments.conf as deposit_conf
+from pcapi.core.users import models as users_models
+from pcapi.repository import repository
+from pcapi.repository import transaction
+
+
+logger = logging.getLogger(__name__)
+
+
+def has_celebrated_their_birthday_since_activation(user: users_models.User) -> bool:
+    return user.deposit_activation_date is not None and user.deposit_activation_date.date() <= user.latest_birthday
+
+
+def has_been_recredited(user: users_models.User) -> bool:
+    if len(user.deposit.recredits) == 0:
+        return False
+
+    if user.deposit.recredits[0].recreditType != deposit_conf.RECREDIT_TYPE_AGE_MAPPING[user.age]:
+        return False
+
+    return True
+
+
+def recredit_underage_users() -> None:
+    sixteen_years_ago = datetime.now() - relativedelta(years=16)
+    eighteen_years_ago = datetime.now() - relativedelta(years=18)
+
+    users: list[users_models.User] = (
+        users_models.User.query.filter(users_models.User.has_underage_beneficiary_role)
+        .filter(users_models.User.dateOfBirth > eighteen_years_ago)
+        .filter(users_models.User.dateOfBirth <= sixteen_years_ago)
```

Comment:
Je ne suis peut-être pas bien réveillé, mais ces 2 filtres ne peuvent jamais être vrais ensemble, non ? Est-ce qu'on ne voulait pas plutôt écrire :

```suggestion
        .filter(users_models.User.dateOfBirth >= sixteen_years_ago)
        .filter(users_models.User.dateOfBirth < eighteen_years_ago)
```

?

Reply:
Je me suis aussi fais avoir au début, mais c'est bien ça 
![image](https://user-images.githubusercontent.com/14235661/140942249-8f59271b-2424-42f5-b35f-527524cb4744.png)

Reply:
Hum, c'est tout à fait correct et si j'avais fait ce dessin je m'en serais rendu compte tout seul. Désolé pour le bruit ! :)